wandb/apis/public/artifacts.py

```diff
@@ -318,12 +319,16 @@ def __init__(
         self.client = client
         self.entity = entity
         self.project = project
-        self.name = name
-        self.type = type
+        self._name = name
+        self._saved_name = name
+        self._type = type
+        self._saved_type = type
         self._attrs = attrs
-        if self._attrs is None:
-            self.load()
+        self.load()
         self._aliases = [a["node"]["alias"] for a in self._attrs["aliases"]["edges"]]
+        self._description = self._attrs["description"]
+        self._tags = [a["node"]["name"] for a in self._attrs["tags"]["edges"]]
+        self._saved_tags = copy(self._tags)
```

Comment:
suggestion(optional): `list` has a builtin `.copy()` method which does the same as `copy.copy` i.e. returns a shallow copy.  Definitely not a big deal, but it'd save the extra import (`from copy import copy` ☝️ )
```suggestion
        self._saved_tags = self._tags.copy()
```